src/rt/memset.d

```diff
@@ -24,6 +24,36 @@ extern (C)
 
 extern (C):
 
+/**
+Set all elements of an array to be a specified value.
+
+Meant to replace _memset16, _memset32, _memset64, _memset80, and _memset128,
+which are not linked in with -betterC, see https://issues.dlang.org/show_bug.cgi?id=19946
+
+Params:
+    array = array to set
+    value = value to set each array element to
+*/
+void _d_memset(T)(T[] array, T value)
+{
+    foreach (i; 0 .. array.length)
+        array[i] = value;
```

Comment:
Why this inefficient assignment? Please fallback to the actual memset

Reply:
The idea is to start with something simple that works. `memset` doesn't support multi-byte values and is not available at CTFE.

Reply:
> The idea is to start with something simple that works. `memset` doesn't support multi-byte values and is not available at CTFE.

Ah sure, I understand. There's already intrinsics (below) to do multi-byte assignment, tho. But yeah, I agree that the complete version should be done separately as it might require a more complex/complete implementation.

I'll leave a list of things we need to account for:
- ellaborate assignment and CTFE (operator overloading, this implementation)
- single byte assignment (straight memset call)
- multi byte assignment (possibly a generic and inline assembly versions, would be cool to have a generic version with the `__vector` API)

For the byte assignments we need to check for immutable qualifiers or for copy ctors.

Reply:
The compiler will automatically do this.

No one cares or should care about dmd performance, LLVM and GCC both go to some length to find these kinds of assignments and optimize them.

Reply:
KIFSS

Reply:
> The compiler will automatically do this.
> 
> No one cares or should care about dmd performance, LLVM and GCC both go to some length to find these kinds of assignments and optimize them.

It does for single-byte assignment, although, for multiple bytes like 32-bit size, it does a poor job. See https://godbolt.org/z/994n163ao . I would also value not having such performance regression on DMD tho. We can probably do a cleaner generic implementation with the `__vector` API, but I agree doing those on separate PRs. I believe there is a huge performance benefit here. This is also a cool thing to propose on the LLVM optimization/transformation passes. I'm not sure how about cache/branch predictor would behave with this inline asm but I think it doesn't matter, given the generated assembly.